Checks

achalco94 opened the pull request:
This is the pull request you asked for Dan

achalco94 committed:
Create ttt_game.py

achalco94 committed:
Update ttt_game.py

achalco94 committed:
Update ttt_game.py

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+player_choice(start)
+counter = 0 
+
+while counter < 4:  
```
indentation is wrong in this entire while loop

DanFlannel replied:
is this even a method? if so please make it one

DanFlannel replied:
do this part towards the end. focus on other things first

DanFlannel replied:
while !isBoardEmpty():

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+while counter < 4:  
+
+    # This while loop will start the game and the while loop will remain for the remainder of the total number of possible turns that the game can be allowed to go for 
+    play1 = input('Player 1 please input the location of where you would want your mark to be')
```
make the 'Player 1 please input the location of where you would want your mark to be' its own variable and a constant

DanFlannel replied:
you can't make constants in python but please look these up as they are important

DanFlannel replied:
make the string inside the input be it's own variable

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+    board[play_1]=(player1)
+    #This allows Player 1 to choose where in the board they would like to place their respective marker on since the input is a string the variable play_1 would take the str input and convert it to a int and change the list accordingly
+
+    play2= input('Player 2 please input the location of where you would want your maker to be')
```
also make the string in here it's own variable

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+## Alex Chalco Tic Tac Toe game please let me know if there is anything wrong with the code pls and thankies 
+
+def the_board(board):
```
rename method please, the_board is not descriptive of what this method is actually doing

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+    print(board[6] + '|' + board[7] + '|' + board[8])
+
+
+def player_choice(choice): #This funtion will allow the user to choose whether Player 1 wants to be either X or O and depending on the first players input it will assign the variable player 1 and player 2 accordingly 
```
put the comment above the method not in the same line

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+
+board=['','','','','','','','','']
+#the_board(board)
+start = input('Player 1 Please choose if you want to be X or O(keep in mind that the input must be either capital X or capital O)')
```
Constant variable please

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+    board[play_2]=(player2)
+    #This allows the same the same thing to happen for player 2 
+
+    the_board(board)
```
put the comment above the method call

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+## Alex Chalco Tic Tac Toe game please let me know if there is anything wrong with the code pls and thankies 
```
header format:
name
file name and creation date
file summary
then pls review, thankies

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+    # Checks for win conditions/ conditions to continue 
+
+    if board[1] == board[2] == board[3]:
+        print('Congrats! Player {} you have won the game'.format(board[1]))
```
definitely make the congrats a constant (yes you can pass in string to print and still format them)

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+    elif board[1] == board[4] == board[7]:
+        print('Congrats! Player {} you have won the game'.format(board[1]))
+        break
+    elif board[1] == board[5] == board[9]:
```
now hear me out..... a  method to check columns? and pass the current state of the board in? returns a bool?

achalco94 replied:
I hear yeah so for example

def column_checker():

and then copy paste why whole if statements except it returns true or false ?
or create a complete new way of checking this out ?

DanFlannel replied:
see above for checking rows to figure out if we can do columns

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+    elif board[3] == board[5] == board[7]:
+        print('Congrats! Player {} you have won the game'.format(board[3]))
+        break 
+    elif counter == 4: 
```
counter number should be a constant? OR we check to see if there are any available spots open left in the board?

achalco94 replied:
I did a constant counter number because I thought that the max possible rounds would be 4ish ?
 either one wins or someone stalls out

DanFlannel replied:
see above to get an idea on how to see if there is an open space available

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+    # This while loop will start the game and the while loop will remain for the remainder of the total number of possible turns that the game can be allowed to go for 
+    play1 = input('Player 1 please input the location of where you would want your mark to be')
+    play_1= int(play1)
+    board[play_1]=(player1)
```
I could keep passing in the value of 1 forever can't I? infinite game! probably should check that here?

achalco94 replied:
so add a check to make sure that the number hasnt been played already then right ?

DanFlannel replied:
yup

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+
+    play2= input('Player 2 please input the location of where you would want your maker to be')
+    play_2 = int(play2)
+    board[play_2]=(player2)
```
also check if this is a valid move here, what if I just take player 2s spots!

achalco94 replied:
Super valid point 
I was honestly hoping scouts honor would go a long way here :(

DanFlannel replied:
creating an isValid function should be easy because you're already storing the places someone has moved

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+    # The following will be what determines the following rounds to come to be
+    # Checks for win conditions/ conditions to continue 
+
+    if board[1] == board[2] == board[3]:
```
outta curiosity............ don't arrays start at 0?

achalco94 replied:
Yeah that was my bad I went back and fixed this

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+    if board[1] == board[2] == board[3]:
+        print('Congrats! Player {} you have won the game'.format(board[1]))
+        break
+    elif board[4] == board[5] == board[6]:
```
interesting .... what if you checked each row of the board in a for loop? and then used a method and pass current state of the board in?

achalco94 replied:
This is something im having a bit of trouble visualizing I dont know how to check for rows  which is why I went the route of just manually checking

DanFlannel replied:
you should be using a 2 dimensional array aka board[x][y] so you can do
```

for(i = 0, i < board.length, i++):
    player = board[i][0]
   count = 0
    for(n = 0, n < board[i].length, n++):
        if(board[i][n] !== player) : break
        else: count++
    if(count == board[i].length) : return true
return false
```

DanFlannel replied:
code above might not work, was writing it on github

achalco94 replied:
ahhhhh i see you point now 
so the way I was going about checking everything was not through a 2d array.... it was actually a blank list and as the game keeps going by the while loop will fill in the list and then I run the elif statements as checks to see if the game has reached an end or not that why I was having trouble visualizing the checking rows thing

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+
+print('Thanks for playing!')
+
+print('This line was created for the sole purpose of making a pull request dont ask me why')
```
donde estas el método principal

DanFlannel replied:
I know some Spanish.......

DanFlannel replied:
also do this last (look up what a main method is in python)

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,79 @@
+## Alex Chalco Tic Tac Toe game please let me know if there is anything wrong with the code pls and thankies 
+
+def the_board(board):
+    #clear_output()
```
where is this clear_output method that we have commented out?

DanFlannel replied:
just saying there should be a board size variable..... would help for a two dimensional array

DanFlannel commented:
Alright, decent first attempt at coding here. If you have questions about how to do things please let me know. Comment on my comments if you questions or question my thoughts (totally allowed to ask me why). HMU, I didn't write any code recommendations for ya here because I want you to try to figure it out, please work on this in small chunks and commit pieces of it, if you resolve a few comments, (hit the resolve conversation button), commit that code and I will review those pieces. 

Missing testing plan and screen shots of games played XD XD (sorry I won't go this hard)
Also summary
Also naming of pull request should be more explicit

I'm a monster I'll stop now

achalco94 committed:
Update ttt_game.py

I have created the 2D array in board and also changed the game board set up please let me know if there is anything wrong that I did

DanFlannel commented on ttt_game.py:
```diff
@@ -0,0 +1,80 @@
+    print('-----')
+    print(board[1][0] + '|' + board[1][1] + '|' + board[1][2])
+    print('-----')
+    print(board[2][0] + '|' + board[2][1] + '|' + board[2][2])
```
use a for loop please for printing these things out

DanFlannel reviewed (changes requested):
you have a 2d array now please do 2d array things

DanFlannel committed:
Create ttt_game_old.py
